qiskit/circuit/quantumcircuit.py

```diff
@@ -47,7 +48,14 @@
 from qiskit.utils.deprecation import deprecate_func
 from . import _classical_resource_map
 from ._utils import sort_parameters
-from .classical import expr
+from .controlflow import ControlFlowOp
+from .controlflow.break_loop import BreakLoopOp, BreakLoopPlaceholder
+from .controlflow.continue_loop import ContinueLoopOp, ContinueLoopPlaceholder
+from .controlflow.for_loop import ForLoopOp, ForLoopContext
+from .controlflow.if_else import IfElseOp, IfContext
+from .controlflow.switch_case import SwitchCaseOp, SwitchContext
+from .controlflow.while_loop import WhileLoopOp, WhileLoopContext
```

Comment:
Yeah exactly - once an `isinstance(x, ControlFlowOp)` checked moved into `append`, I didn't want to be doing runtime imports within it any more, since I don't want control-flow to cause slowdown if it's not being used.  Control-flow and the circuit are fairly naturally mutually dependent as things stand, so there's always going to be a bit of weirdness.  I think this way round actually ends up with fewer runtime imports, though, so it's probably better anyway.